src/gui/widgets/SideBar.cpp

```diff
@@ -107,53 +107,56 @@ SideBar::~SideBar()
 
 
 
-void SideBar::appendTab( SideBarWidget * _sbw )
+void SideBar::appendTab( SideBarWidget *widget )
 {
-	SideBarButton * btn = new SideBarButton( orientation(), this );
-	btn->setText( _sbw->title() );
-	btn->setIcon( _sbw->icon() );
-	btn->setCheckable( true );
-	m_widgets[btn] = _sbw;
-	m_btnGroup.addButton( btn );
-	addWidget( btn );
-
-	_sbw->hide();
-	_sbw->setMinimumWidth( 200 );
-
-	ToolTip::add( btn, _sbw->title() );
+	SideBarButton *button = new SideBarButton( orientation(), this );
+	button->setText( widget->title() );
+	button->setIcon( widget->icon() );
+	button->setCheckable( true );
+	m_widgets[button] = widget;
+	m_btnGroup.addButton( button );
+	addWidget( button );
+
+	widget->hide();
+	widget->setMinimumWidth( 200 );
+
+	ToolTip::add( button, widget->title() );
 }
 
 
 
 
-void SideBar::toggleButton( QAbstractButton * _btn )
+void SideBar::toggleButton( QAbstractButton * button )
 {
-	QToolButton * toolButton = NULL;
-	QWidget * activeWidget = NULL;
-	for( ButtonMap::Iterator it = m_widgets.begin();
-							it != m_widgets.end(); ++it )
+	QToolButton *toolButton = NULL;
+	QWidget *activeWidget = NULL;
+
+	for( auto it = m_widgets.begin(); it != m_widgets.end(); ++it )
 	{
```

Comment:
On 15-01-20 06:24 AM, Lukas W wrote:
>> +	for( auto it = m_widgets.begin(); it != m_widgets.end(); ++it )
>>  	{
> 
> I think we agreed on using C++11's `:` iterator and avoiding to use the 
> `auto` keyword.

The `for :` syntax only iterates over the values of the map, but we're using the keys and values. Or are you thinking of something like this?
```cpp
for( QToolButton *curBtn : m_widgets.keys() )
{
	QWidget *curWidget = m_widgets[curBtn];
```